Error generating file ATen_generated_THCBlas.cu.o

I am trying to build PyTorch from source on macOS High Sierra Version 10.13.3 with an NVIDIA GeForce GT 750M. I'm following the instructions at https://github.com/pytorch/pytorch#from-source. `cat ~/.bash_profile` prints
```
export PATH=/Developer/NVIDIA/CUDA-9.1/bin${PATH:+:${PATH}}
export DYLD_LIBRARY_PATH=/Developer/NVIDIA/CUDA-9.1/lib\
                         ${DYLD_LIBRARY_PATH:+:${DYLD_LIBRARY_PATH}}
export  DYLD_LIBRARY_PATH=/usr/local/cuda/lib:$DYLD_LIBRARY_PATH

# added by Anaconda3 5.0.1 installer
export PATH="/Users/carlos/anaconda3/bin:$PATH"
export CMAKE_PREFIX_PATH=/Users/carlos/anaconda3
```

`nvcc --version` prints
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Tue_Dec_19_21:36:29_CST_2017
Cuda compilation tools, release 9.1, V9.1.128
```
`gcc --version` prints 
```
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
`cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2` prints
```
#define CUDNN_MAJOR 7
#define CUDNN_MINOR 0
#define CUDNN_PATCHLEVEL 5
```
When I enter `MACOSX_DEPLOYMENT_TARGET=10.13 CC=clang CXX=clang++ python3 setup.py install`, as per the instructions, I receive the following error message:
```
python3 setup.py install
running install
running build_deps
CMake Warning at /Users/carlos/anaconda3/share/cmake-3.9/Modules/Platform/Darwin-Initialize.cmake:121 (message):
  Ignoring CMAKE_OSX_SYSROOT value:

   /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk

  because the directory does not exist.
Call Stack (most recent call first):
  /Users/carlos/anaconda3/share/cmake-3.9/Modules/CMakeSystemSpecificInitialize.cmake:8 (include)
  CMakeLists.txt:9 (project)


-- Automatic GPU detection failed. Building for common architectures.
-- Autodetected CUDA architecture(s): 3.0;3.5;5.0;5.2;6.0;6.1;6.1+PTX
-- Found CUDA with FP16 support, compiling with torch.CudaHalfTensor
-- Removing -DNDEBUG from compile flags
-- MAGMA not found. Compiling without MAGMA support
-- Could not find hardware support for NEON on this machine.
-- No OMAP3 processor on this machine.
-- No OMAP4 processor on this machine.
-- SSE2 Found
-- SSE3 Found
-- AVX Found
-- AVX2 Found
-- Atomics: using GCC intrinsics
-- Checking for [mkl_intel_lp64 - mkl_intel_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_intel_lp64: /Users/carlos/anaconda3/lib/libmkl_intel_lp64.dylib
--   Library mkl_intel_thread: /Users/carlos/anaconda3/lib/libmkl_intel_thread.dylib
--   Library mkl_core: /Users/carlos/anaconda3/lib/libmkl_core.dylib
--   Library iomp5: /Users/carlos/anaconda3/lib/libiomp5.dylib
--   Library pthread: /usr/lib/libpthread.dylib
--   Library m: /usr/lib/libm.dylib
-- MKL library found
-- Found a library with BLAS API (mkl).
-- Found a library with LAPACK API. (mkl)
-- Found cuDNN: v7.0.5  (include: /usr/local/cuda/include, library: /usr/local/cuda/lib/libcudnn.7.dylib)
-- Could NOT find NNPACK (missing: NNPACK_INCLUDE_DIR NNPACK_LIBRARY CPUINFO_LIBRARY PTHREADPOOL_LIBRARY) 
-- NNPACK not found. Compiling without nNPACK support
CMake Deprecation Warning at src/ATen/CMakeLists.txt:7 (CMAKE_POLICY):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Using python found in /Users/carlos/anaconda3/bin/python3
disable contrib because ATEN_NO_CONTRIB is set
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/carlos/downloads/pytorch/torch/lib/build/aten
[  0%] Built target aten_files_are_generated
[  0%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o
[  0%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorConv.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorSort.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorTypeUtils.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorIndex.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorScatterGather.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorTopK.cu.o
[  2%] Building NVCC (Device) object src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCTensorRandom.cu.o
/Users/carlos/downloads/pytorch/aten/src/THC/THCBlas.cu(349): error: argument of type "long *" is incompatible with parameter of type "int64_t *"

/Users/carlos/downloads/pytorch/aten/src/THC/THCBlas.cu(349): error: argument of type "long *" is incompatible with parameter of type "int64_t *"

/Users/carlos/downloads/pytorch/aten/src/THC/THCBlas.cu(349): error: argument of type "long *" is incompatible with parameter of type "int64_t *"

3 errors detected in the compilation of "/var/folders/sg/mtsds5v51394m14s124_60xc0000gn/T//tmpxft_000021c8_00000000-11_THCBlas.compute_61.cpp1.ii".
CMake Error at ATen_generated_THCBlas.cu.o.cmake:267 (message):
  Error generating file
  /Users/carlos/downloads/pytorch/torch/lib/build/aten/src/ATen/CMakeFiles/ATen.dir/__/THC/./ATen_generated_THCBlas.cu.o


make[2]: *** [src/ATen/CMakeFiles/ATen.dir/__/THC/ATen_generated_THCBlas.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

--- a/aten/src/THC/THCBlas.cu
+++ b/aten/src/THC/THCBlas.cu
@@ -335,9 +335,9 @@ void THCudaBlas_Dgemm(THCState *state, char transa, char transb, int64_t m, int6
 }
 
 #if CUDA_VERSION >= 9010
-void THCudaBlas_HgemmStridedBatched(THCState *state, char transa, char transb, long m, long n, long k,
-                             half alpha, const half *a, long lda, long strideA, const half *b, long ldb, long strideB,
-                             half beta, half *c, long ldc, long strideC, long batchCount)
+void THCudaBlas_HgemmStridedBatched(THCState *state, char transa, char transb, int64_t m, int64_t n, int64_t k,
+                             half alpha, const half *a, int64_t lda, int64_t strideA, const half *b, int64_t ldb, int64_t strideB,
+                             half beta, half *c, int64_t ldc, int64_t strideC, int64_t batchCount)
 {
   if( (m >= INT_MAX) || (n >= INT_MAX) || (k >= INT_MAX) || (lda >= INT_MAX)  || (ldb >= INT_MAX) || (ldc >= INT_MAX) || (batchCount >= INT_MAX) )
 

--- a/aten/src/THC/THCBlas.h
+++ b/aten/src/THC/THCBlas.h
@@ -41,9 +41,9 @@ THC_API void THCudaBlas_DgemmStridedBatched(THCState *state, char transa, char t
 #endif
 
 #if CUDA_VERSION >= 9010
-void THCudaBlas_HgemmStridedBatched(THCState *state, char transa, char transb, long m, long n, long k,
-                                     half alpha, const half *a, long lda, long strideA, const half *b, long ldb, long strideB,
-                                                                  half beta, half *c, long ldc, long strideC, long batchCount);
+void THCudaBlas_HgemmStridedBatched(THCState *state, char transa, char transb, int64_t m, int64_t n, int64_t k,
+                                     half alpha, const half *a, int64_t lda, int64_t strideA, const half *b, int64_t ldb, int64_t strideB,
+                                                                  half beta, half *c, int64_t ldc, int64_t strideC, int64_t batchCount);
 #endif
 
 /* Inverse */

--- a/aten/src/THC/generic/THCTensorMathBlas.cu
+++ b/aten/src/THC/generic/THCTensorMathBlas.cu
@@ -438,8 +438,8 @@ __global__ void createBatchGemmBuffer(const real** buffer, real* data,
 }
 
 __global__ void createBatchGemmBuffer3(const real** buffer1, const real ** buffer2, const real ** buffer3, real* data1,
-                                       real * data2, real * data3, long stride1, long stride2, long stride3, long num_batches) {
-  const long idx = blockIdx.x * blockDim.x + threadIdx.x;
+                                       real * data2, real * data3, int64_t stride1, int64_t stride2, int64_t stride3, int64_t num_batches) {
+  const int64_t idx = blockIdx.x * blockDim.x + threadIdx.x;
   if (idx < num_batches) {
     buffer1[idx] = data1 + idx * stride1;
     buffer2[idx] = data2 + idx * stride2;
@@ -671,7 +671,7 @@ THCTensor_(baddbmm)(THCState *state, THCTensor *result, real beta, THCTensor *t,
       THCTensor_(data)(state, result_), ldc, result_->stride[0],
       num_batches);
    } else {
-      for (long i = 0; i < num_batches; ++i) {
+      for (int64_t i = 0; i < num_batches; ++i) {
         THCudaBlas_Hgemm(
         state,
         transpose_batch1,

